components/script/script_module.rs

```diff
@@ -758,20 +759,21 @@ impl ModuleTree {
 
                 // Step 8.
 
-                for url in urls {
+                let visited_urls = self.visited_urls.borrow().clone();
+                let options = options.descendant_fetch_options();
+
+                for url in urls_to_fetch {
                     // https://html.spec.whatwg.org/multipage/#internal-module-script-graph-fetching-procedure
                     // Step 1.
-                    assert!(visited_urls.get(&url).is_some());
-
-                    let options = options.descendant_fetch_options();
+                    debug_assert!(self.visited_urls.borrow().contains(&url));
```

Comment:
```suggestion
                    assert!(self.visited_urls.borrow().contains(&url));
```

Reply:
Are we using assert for all places that has a potential GC hazard? Even after making changes to ensure safety?

Reply:
The original code used assert, so we shouldn't modify it in this PR. The specification this code is implementing explicitly has an assert step.